horovod/common/operations.cc

```diff
@@ -659,7 +674,58 @@ void set_int_from_env(const char* env, int& val) {
 //      otherwise we may end up dispatching many blocked threads and never make
 //      progress if we have a thread pool limit.
 bool RunLoopOnce(HorovodGlobalState& state, MPIContext& ctx, bool is_coordinator);
+
+#if HAVE_MLSL
+static void server_affinity_set(int affinity) {
+  cpu_set_t cpuset;
+  pthread_t current_thread = pthread_self();
+
+  __CPU_ZERO_S(sizeof(cpu_set_t), &cpuset);
+  __CPU_SET_S(affinity, sizeof(cpu_set_t), &cpuset);
+
+  if (pthread_setaffinity_np(current_thread, sizeof(cpu_set_t), &cpuset) != 0)
+      perror("setaffinity failed\n");
+
+  // Check if we set the affinity correctly
+  if (pthread_getaffinity_np(current_thread, sizeof(cpu_set_t), &cpuset) != 0)
+      perror("sched_getaffinity failed\n");
+
+  for (int core_idx = 0; core_idx < __CPU_SETSIZE; core_idx++)
+       if (__CPU_ISSET_S(core_idx, sizeof(cpu_set_t), &cpuset))
+           LOG(DEBUG) << "BG-thread affinity " << core_idx;
+}
+#endif
 void BackgroundThreadLoop(HorovodGlobalState& state, MPIContext& ctx) {
+#if HAVE_MLSL
+  char* hvd_mlsl_bg_thread_env = NULL;
+  int bg_thread_affinity = 0;
+  if ((hvd_mlsl_bg_thread_env = getenv("HVD_MLSL_BGT_AFFINITY")) != NULL)
```

Comment:
Please add environment variable to `operations.h` and rename it to `HOROVOD_MLSL_BGT_AFFINITY`